streams/src/main/java/org/apache/kafka/streams/state/internals/MeteredWindowStore.java

```diff
@@ -217,6 +230,24 @@ public V fetch(final K key,
                                         final long timeFrom,
                                         final long timeTo) {
         Objects.requireNonNull(key, "key cannot be null");
+
+        if (wrapped().persistent()) {
+
+            final long actualWindowStartTime = getActualWindowStartTime(timeFrom);
+
+            if (timeTo < actualWindowStartTime) {
+                return KeyValueIterators.emptyWindowStoreIterator();
+            }
+
+            return new MeteredWindowStoreIterator<>(
+                wrapped().fetch(keyBytes(key), actualWindowStartTime, timeTo),
+                fetchSensor,
+                streamsMetrics,
+                serdes,
+                time
+            );
```

Comment:
There are many duplicated codes for this section, do you think it'd better to make them a separate method?

Reply:
For the most part, only getting the actual start time and returning empty iterator are duplicate. The MeteredWindowStoreIterator object is built using a different method call every time. Do you think the first part needs to be abstracted out to a different method? That might also differ across different methods..